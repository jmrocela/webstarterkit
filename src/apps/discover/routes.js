import React from 'react';
import browserHistory from 'react-router/lib/browserHistory';
import IndexRoute from 'react-router/lib/IndexRoute';
import Route from 'react-router/lib/Route';
import Router from 'react-router/lib/Router';
import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import { routerMiddleware, routerReducer } from 'react-router-redux';
import { canusedom } from '../../client/helpers/domutil';
import thunk from 'redux-thunk';

import Wrapper from '../../client/components/wrapper';
import HomeContainer from './containers/home_container';
import NotFoundContainer from './containers/notfound_container';

import AppReducer from '../../client/redux/app';

export const configureStore = (initial = {}) => {
  const store = createStore(
    combineReducers({
      routing: routerReducer,
      app: AppReducer
    }),
    initial,
    compose(
      applyMiddleware(thunk, routerMiddleware(browserHistory)),
      typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f
    )
  );

  if (module.hot) {
    module.hot.accept();
  }

  return store;
};

if (canusedom) {
  browserHistory.listenBefore(location => {
    // record url change
  });
}

export const routes = (
  <Router history={browserHistory}>
    <Route path='/' component={Wrapper}>
      <IndexRoute component={HomeContainer} />
    </Route>
    <Route path='*' component={NotFoundContainer} />
  </Router>
);

export default routes;
