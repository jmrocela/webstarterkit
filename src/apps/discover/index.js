import React from 'react';
import ReactDOM from 'react-dom';
import match from 'react-router/lib/match';
import Router from 'react-router/lib/Router';
import browserHistory from 'react-router/lib/browserHistory';
import { Provider } from 'react-redux';
import { IntlProvider, addLocaleData } from 'react-intl';

import { configureStore, routes } from './routes';

let devtool;
if (process.env.NODE_ENV !== 'production') {
  devtool = window.devToolsExtension && window.devToolsExtension();
}

const store = configureStore(window.__data, devtool);

const reactRoot = window.document.getElementById('react-root');
match({ history: browserHistory, routes: routes }, (error, redirectLocation, renderProps) => {
  addLocaleData({ locale: window.__locale, pluralRuleFunction: () => {} });
  ReactDOM.render(<IntlProvider locale="en" messages={window.__localeMessages || {}}><Provider store={store}><Router {...renderProps} /></Provider></IntlProvider>, reactRoot);
});

if (process.env.NODE_ENV !== 'production') {
  if (!reactRoot.firstChild || !reactRoot.firstChild.attributes ||
    !reactRoot.firstChild.attributes['data-react-checksum']) {
    console.error('Server-side React render was discarded. Make sure that your initial render does not contain any client-side code.');
  }
}
