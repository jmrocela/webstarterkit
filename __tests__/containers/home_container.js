import React from 'react';
import { configureStore } from '../../src/apps/discover/routes';
import HomeContainer from '../../src/apps/discover/containers/home_container';

const store = configureStore({});

describe('HomeContainer', () => {
  it('should render the home container', () => {
    const wrapper = shallow(<HomeContainer store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});