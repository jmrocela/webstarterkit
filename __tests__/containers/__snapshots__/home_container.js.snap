exports[`HomeContainer should render a label 1`] = `
<HomeContainer
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
